markdown/events/2021/bytesize-4-github-contribution-basics.md

```diff
@@ -34,3 +34,97 @@ This will cover:
+
+A few of things that are of common nomenclature in Git and comments and repositories or short repos. So if you're talking with people in nf-core, they might refer to things like a repo; just to point you towards one, so a repository is basically a project where your code is stored. An example would be the nf-core/rna-seq repository that hosts all the code for the rna pipeline. We have many of these repositories or repos. For example each of the pipelines has a separate one, but we also have repositories that host  code for our web page or some documentation or some test data sets so this is something that is a repository. A set of logical edits are chunked together and can be committed to my repository so that they can be within this version control system.
+
+The git tree of a repository can have branches so my repository might have different separate branches, which is a kind of a way to organize code in the repository, so people can have a look at that later.
+
+That's just a very very brief intro into the basic commands here. We have git clone which means downloader; git add, which means stage changes that I have already made, like adding a new file for example; ready to comment git comment is actually then adding this to the yeah more or less adding this to the repository with a comment. So I can actually comment on that, I can say I’ve added this file which would be quite a nice commit, write a message because that means that people know what I’ve done through this comment. Then I can also git push, which means I can push these changes to the repository. If somebody else made some changes in the web for example on a repository that I actually want to integrate with my local copy of of the repository I can do a git pull, which will pull these from there. There is also some additional vocabulary here that people would like to probably see a bit about; “PR” means pull request and fork makes a derivative copy, so I can fork a repository that is present on github and make a copy for my own that can work on independently, I could continue development on that and never return to the original repository. That happens, people abandon projects quite frequently, but I could also use that to develop my own type of feature - let's say mapping methods, changes to a pipeline or something like that and send them back to the main repository via a pull request. [4:27](https://youtu.be/gTEXDXWf4hE?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=265)
```

Comment:
```suggestion
This slide shows some of the basic git commands. We have `git clone` which means downloader; `git add`, which means stage changes that I have already made, like adding a new file for example; ready to commit; `git commit` is actually then adding this to the yeah more or less adding this to the repository with a comment. So I can actually comment on that, I can say I’ve added this file which would be quite a nice commit, write a message because that means that people know what I’ve done through this comment. Then I can also `git push`, which means I can push these changes to the repository. If somebody else made some changes in the web for example on a repository that I actually want to integrate with my local copy of of the repository I can do a `git pull`, which will pull these from there. There is also some additional vocabulary here that people would like to probably see a bit about; “PR” means pull request and fork means a derivative copy, so I can fork a repository that is present on Github and make a copy for my own that I can work on independently, I could continue development on that and never return to the original repository. That happens, people abandon projects quite frequently, but I could also use that to develop my own type of feature - let's say mapping methods, changes to a pipeline or something like that and send them back to the main repository via a pull request. [4:27](https://youtu.be/gTEXDXWf4hE?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=265)
```